cli/src/pcluster/validators/slurm_settings_validator.py

```diff
@@ -144,3 +144,36 @@ def _validate(self, custom_settings: List[Dict], include_file_url: str):
                 "under SlurmSettings.",
                 FailureLevel.ERROR,
             )
+
+
+class SlurmNodePrioritiesWarningValidator(Validator):
+    """
+    Slurm Node Weights Warning Validator.
+
+    This validator checks, within a queue, whether any dynamic nodes have lower node weights than any static
+    nodes and throws a warning if that's the case.
+    """
+
+    def _validate(self, queue_name: str, compute_resources: List[Dict]):
+        static_priorities = {cr.name: cr.static_node_priority for cr in compute_resources}
+        dynamic_priorities = {cr.name: cr.dynamic_node_priority for cr in compute_resources}
+
+        bad_static_priorities = {
+            key: value
+            for key, value in static_priorities.items()
+            if value in range(min(dynamic_priorities.values()), max(dynamic_priorities.values()) + 1)
+        }
+        bad_dynamic_priorities = {
+            key: value
+            for key, value in dynamic_priorities.items()
+            if value in range(min(static_priorities.values()), max(static_priorities.values()) + 1)
+        }
+
+        if bad_static_priorities or bad_dynamic_priorities:
+            self._add_failure(
+                f"Some compute resources in queue {queue_name} have static nodes with higher or equal priority than "
+                f"other dynamic nodes in the same queue. "
+                f"Possible problematic static node priorities are {bad_static_priorities}. "
```

Comment:
Following the previous comment, we can enrich this message reporting also the boundaries extracted above (max_static and min_dynamic). This would make it easier to review the "offending" values.

Reply:
The current message is already providing this information, although not explicitly as you are saying.

What is being printed at the moment are all the values of priority (both static and dynamic) that go against the "soft rule" of having all dynamic priorities higher than the static ones within the same queue. So the max static and min dynamic are already included in the printed values (they will be respectively the last value of static priority and the first value of dynamic priority being printed in the message).

Reply:
Yes. I saw the examples in the tests.
However if you have a long list of Compute Resources is may be confusing and not immediately obvious which value you have to pick from which list..because you have to mentally rebuild the logic behind the check.

While in the other way you can immediately tell the customers which is the "offending threshold" in both cases.
Also you can avoid defining the full range, since what you need is only the min or the max of the range .

Anyway, I'll approve the PR since it is a sort of Nitpick.